zio-http/src/main/scala/zhttp/service/Server.scala

```diff
@@ -129,13 +139,14 @@ object Server {
     acceptContinue: Boolean = false,
     keepAlive: Boolean = false,
     consolidateFlush: Boolean = false,
+    allocator: Option[PooledByteBufAllocator] = None,
     flowControl: Boolean = true,
   )
 
   /**
    * Holds server start information.
    */
-  final case class Start(port: Int = 0)
+  final case class Start(port: Int = 0, allocator: Option[PooledByteBufAllocator] = None)
```

Comment:
```suggestion
  final class Start(port: Int = 0, private val allocator: Option[PooledByteBufAllocator] = None)
```

Reply:
- Keep allocator in Start, and add helper methods to access internal metrics.
- Can we set allocator with the default allocator for the server, such that the API doesn't return an `Option`?

Reply:
* Done
* Yes, done the change.